conans/client/rest/uploader_downloader.py

```diff
@@ -52,44 +48,51 @@ def upload(self, url, abs_path, auth=None, dedup=False, retry=None, retry_wait=N
             if response.status_code == 201:  # Artifactory returns 201 if the file is there
                 return response
 
-        if not self.output.is_terminal:
-            self.output.info("")
-        # Actual transfer of the real content
-        it = load_in_chunks(abs_path, self.chunk_size)
-        # Now it is a chunked read file
-        file_size = os.stat(abs_path).st_size
-        file_name = os.path.basename(abs_path)
-        it = upload_with_progress(file_size, it, self.chunk_size, self.output, file_name)
-        # Now it will print progress in each iteration
-        iterable_to_file = IterableToFileAdapter(it, file_size)
-        # Now it is prepared to work with request
         ret = call_with_retry(self.output, retry, retry_wait, self._upload_file, url,
-                              data=iterable_to_file, headers=headers, auth=auth)
-
+                              abs_path=abs_path, headers=headers, auth=auth)
         return ret
 
-    def _upload_file(self, url, data,  headers, auth):
-        try:
-            response = self.requester.put(url, data=data, verify=self.verify,
-                                          headers=headers, auth=auth)
+    def _upload_file(self, url, abs_path,  headers, auth):
 
-            if response.status_code == 400:
-                raise RequestErrorException(response_to_str(response))
-
-            if response.status_code == 401:
-                raise AuthenticationException(response_to_str(response))
-
-            if response.status_code == 403:
-                if auth.token is None:
+        file_size = os.stat(abs_path).st_size
+        file_name = os.path.basename(abs_path)
+        description = "Uploading {}".format(file_name)
+
+        def load_in_chunks(file, size):
+            """Lazy function (generator) to read a file piece by piece.
+            Default chunk size: 1k."""
+            while True:
+                chunk = file.read(size)
+                if not chunk:
+                    break
+                yield chunk
+
+        with open(abs_path, mode='rb') as file_handler:
+            progress = progress_bar.ReadProgress(file_size, self.output, description)
```

Comment:
Is it possible that this progress bar is not being "closed"? I cannot see where it calls the ``close()`` method.

Reply:
I wonder how much different is this progress bar to the ``with progress_bar.open_binary()`` one. Not to do in this PR, but seems that the same could be used here too.

Reply:
The bar should be closed when the iteration is finished:
https://github.com/conan-io/conan/blob/80d1e348cb82b1fed578a557a4297c70d528e85e/conans/util/progress_bar.py#L88
in the update method of the ReadProgress bar